### PR TITLE
Pulsar 2.9.3 release notes

### PR DESCRIPTION
**Pull request by mattisonchao**
image

image

image

### CODE REVIEW

File: site2/website-next/release-notes/versioned/client-cpp-2.9.3.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+id: client-cpp-2.9.3
+title: Client CPP 2.9.3
+sidebar_label: Client CPP 2.9.3
+---
+
+- [fix][c++] Fix UnknownError might be returned for a partitioned producer[15161](https://github.com/apache/pulsar/pull/15161)
```
Comment (Anonymitaet) [inline site2/website-next/release-notes/versioned/client-cpp-2.9.3.md:7]:
```suggestion
- [fix][c++] Fix UnknownError might be returned for a partitioned producer [#15161](https://github.com/apache/pulsar/pull/15161)
```

Example: https://pulsar.apache.org/release-notes/versioned/pulsar-2.10.0

File: site2/website-next/release-notes/versioned/client-java-2.9.3.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+id: client-java-2.9.3
+title: Client Java 2.9.3
+sidebar_label: Client Java 2.9.3
+---
+
+- [fix][java] Fix ConsumerBuilderImpl can not set null to deadLetterPolicy.[14980](https://github.com/apache/pulsar/pull/14980)
```
Comment (Anonymitaet) [inline site2/website-next/release-notes/versioned/client-java-2.9.3.md:7]:
```suggestion
- [fix][java] Fix ConsumerBuilderImpl can not set null to deadLetterPolicy [#14980](https://github.com/apache/pulsar/pull/14980)
```
same comments for all occurrences

File: site2/website-next/release-notes/versioned/client-python-2.9.3.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+id: client-python-2.9.3
+title: Client Python 2.9.3
+sidebar_label: Client Python 2.9.3
+---
+
+- [fix][python] Fix generated Python protobuf code not compatible with latest protobuf package[15846](https://github.com/apache/pulsar/pull/15846)
+- [improve][python] Added build for Arm64 on Linux and Python 3.10[15004](https://github.com/apache/pulsar/pull/15004)
```
Comment (Anonymitaet) [inline site2/website-next/release-notes/versioned/client-python-2.9.3.md:8]: same use `past tense` while others use `present tense`, please keep consistent

**Review by Anonymitaet — COMMENTED** [review]
@momo-jun FYI

File: site2/website-next/release-notes/versioned/client-cpp-2.9.3.md (hunk 1 above)
Comment (mattisonchao) [reply site2/website-next/release-notes/versioned/client-cpp-2.9.3.md]: fixed.

File: site2/website-next/release-notes/versioned/client-python-2.9.3.md (hunk 3 above)
Comment (mattisonchao) [reply site2/website-next/release-notes/versioned/client-python-2.9.3.md]: fixed

File: site2/website-next/release-notes/versioned/client-java-2.9.3.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+sidebar_label: Client Java 2.9.3
+---
+
+- [fix][java] Fix ConsumerBuilderImpl can not set null to deadLetterPolicy [#14980](https://github.com/apache/pulsar/pull/14980)
+- [fix][java] Fix conversion of `TimestampMillisConversion` has no effect when Jsr310Conversion enabled [#15863](https://github.com/apache/pulsar/pull/15863)
+- [fix][java] Fix messages sent by producers without schema cannot be decoded [#15622](https://github.com/apache/pulsar/pull/15622)
+- [fix][java] Fix performance regression with message listener [#15162](https://github.com/apache/pulsar/pull/15162)
+- [fix][java] Fix potentially unfinished CompletableFuture in doReconsumeLater [#14947](https://github.com/apache/pulsar/pull/14947)
+- [fix][java] Fix race condition in consumer redelivery [#14687](https://github.com/apache/pulsar/pull/14687)
+- [fix][java] Fix the producer OOM if got exception while add message to batch container [#12170](https://github.com/apache/pulsar/pull/12170)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/client-java-2.9.3.md:13]:
```suggestion
- [fix][java] Fix the producer OOM if got an exception while adding messages to batch container [#12170](https://github.com/apache/pulsar/pull/12170)
```

File: site2/website-next/release-notes/versioned/client-java-2.9.3.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+- [fix][java] PartitionsAutoUpdateFuture never complete [#14625](https://github.com/apache/pulsar/pull/14625)
+- [fix][java] Remove consumer when close consumer command is received [#15761](https://github.com/apache/pulsar/pull/15761)
+- [fix][java] Fix PartitionedProducerImpl flushAsync always fail when one partition send TimeOutException [#14602](https://github.com/apache/pulsar/pull/14602)
+- [improve][java] Add pending messages information while print the producer stats [#15440](https://github.com/apache/pulsar/pull/15440)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/client-java-2.9.3.md:17]:
```suggestion
- [improve][java] Add pending messages information while printing the producer stats [#15440](https://github.com/apache/pulsar/pull/15440)
```

File: site2/website-next/release-notes/versioned/client-cpp-2.9.3.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+id: client-cpp-2.9.3
+title: Client CPP 2.9.3
+sidebar_label: Client CPP 2.9.3
+---
+
+- [fix][c++] Fix UnknownError might be returned for a partitioned producer [#15161](https://github.com/apache/pulsar/pull/15161)
+- [fix][c++] Fix connection is not closed when broker closes the connection to proxy [#15009](https://github.com/apache/pulsar/pull/15009)
+- [fix][c++] Fix producer is never destructed until client is closed [#14797](https://github.com/apache/pulsar/pull/14797)
+- [fix][c++] Fix segmentation fault when creating socket failed [#14834](https://github.com/apache/pulsar/pull/14834)
+- [fix][c++] Fix send callback might not be invoked in key based batching [#14898](https://github.com/apache/pulsar/pull/14898)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/client-cpp-2.9.3.md:11]:
```suggestion
- [fix][c++] Fix send callback might not be invoked in key-based batching [#14898](https://github.com/apache/pulsar/pull/14898)
```

File: site2/website-next/release-notes/versioned/client-python-2.9.3.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+---
+id: client-python-2.9.3
+title: Client Python 2.9.3
+sidebar_label: Client Python 2.9.3
+---
+
+- [fix][python] Fix generated Python protobuf code not compatible with latest protobuf package [#15846](https://github.com/apache/pulsar/pull/15846)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/client-python-2.9.3.md:7]:
```suggestion
- [fix][python] Fix generated Python protobuf code not compatible with the latest protobuf package [#15846](https://github.com/apache/pulsar/pull/15846)
```

File: site2/website-next/release-notes/versioned/client-python-2.9.3.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+- [improve][python] Add build for Arm64 on Linux and Python 3.10 [#15004](https://github.com/apache/pulsar/pull/15004)
+- [improve][python] Add build for wheels on Alpine linux [#15016](https://github.com/apache/pulsar/pull/15016)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/client-python-2.9.3.md:9]:
```suggestion
- [improve][python] Add build for wheels on Alpine Linux [#15016](https://github.com/apache/pulsar/pull/15016)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+
+- [fix][connector] Pass client builder if no service url provided to debezium connector [#12145](https://github.com/apache/pulsar/pull/12145)
+- [fix][connector] Throw exceptions when kafka offset backing store failed to start  [#14491](https://github.com/apache/pulsar/pull/14491)
+
+#### Tiered Storage
+
+- [fix][tiered-storage] Fix NoClassDefFoundError: com/google/inject/AbstractModule in pulsar-io/batch-data-generator and Jcloud offloader [#14150](https://github.com/apache/pulsar/pull/14150)
+- [improve][tiered-storage] Add debug information  [#14907](https://github.com/apache/pulsar/pull/14907)
+- [improve][tiered-storage] Add pure S3 provider for the offloader [#15710](https://github.com/apache/pulsar/pull/15710)
+- [improve][tiered-storage] Reduce cpu usage when offloading the ledger [#15063](https://github.com/apache/pulsar/pull/15063)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:194]:
```suggestion
- [improve][tiered-storage] Reduce CPU usage when offloading the ledger [#15063](https://github.com/apache/pulsar/pull/15063)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+- [fix][function] Fix pulsar-managed runtimes failed start function with package URL from package management service [#14814](https://github.com/apache/pulsar/pull/14814)
+- [fix][function] Fix some IOExceptions when create functions from package URL [#14553](https://github.com/apache/pulsar/pull/14553)
+- [fix][function] Handle NPE when `getLeader` returns null [#15058](https://github.com/apache/pulsar/pull/15058)
+- [fix][function] Provide default error handler for function log appender [#15728](https://github.com/apache/pulsar/pull/15728)
+- [fix][function] Refine file io connector [#15250](https://github.com/apache/pulsar/pull/15250)
+
+#### Connector
+
+- [fix][connector] Pass client builder if no service url provided to debezium connector [#12145](https://github.com/apache/pulsar/pull/12145)
+- [fix][connector] Throw exceptions when kafka offset backing store failed to start  [#14491](https://github.com/apache/pulsar/pull/14491)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:187]:
```suggestion
- [fix][connector] Throw exceptions when Kafka offset backing store failed to start  [#14491](https://github.com/apache/pulsar/pull/14491)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+- [fix][function] Check executor null when close the FileSource [#15247](https://github.com/apache/pulsar/pull/15247)
+- [fix][function] Fix pulsar-managed runtimes failed start function with package URL from package management service [#14814](https://github.com/apache/pulsar/pull/14814)
+- [fix][function] Fix some IOExceptions when create functions from package URL [#14553](https://github.com/apache/pulsar/pull/14553)
+- [fix][function] Handle NPE when `getLeader` returns null [#15058](https://github.com/apache/pulsar/pull/15058)
+- [fix][function] Provide default error handler for function log appender [#15728](https://github.com/apache/pulsar/pull/15728)
+- [fix][function] Refine file io connector [#15250](https://github.com/apache/pulsar/pull/15250)
+
+#### Connector
+
+- [fix][connector] Pass client builder if no service url provided to debezium connector [#12145](https://github.com/apache/pulsar/pull/12145)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:186]:
```suggestion
- [fix][connector] Pass client builder if no service URL is provided to Debezium connector [#12145](https://github.com/apache/pulsar/pull/12145)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 12]
```diff
@@ -0,0 +1,213 @@
...
+- [fix][txn] Fix transaction buffer recover BrokerMetadataException close topic [#14709](https://github.com/apache/pulsar/pull/14709)
+- [fix][txn] Fix transaction buffer recover reader and writer fail [#14801](https://github.com/apache/pulsar/pull/14801)
+- [fix][txn] Fix transaction buffer recover throw cursor already close [#14807](https://github.com/apache/pulsar/pull/14807)
+- [fix][txn] Fix transaction log recover throw cursor already close [#14810](https://github.com/apache/pulsar/pull/14810)
+- [fix][txn] Fix transaction pending ack store managed ledger WriteFail state [#14738](https://github.com/apache/pulsar/pull/14738)
+- [fix][txn] Fix transaction producer stuck problem [#15061](https://github.com/apache/pulsar/pull/15061)
+- [fix][txn] Fix transaction component recover fillQueue [#15418](https://github.com/apache/pulsar/pull/15418)
+- [fix][txn] Fix transasction ack batch message [#15875](https://github.com/apache/pulsar/pull/15875)
+- [improve][txn] Avoid create multiple future and exception handler [#15089](https://github.com/apache/pulsar/pull/15089)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:128]:
```suggestion
- [improve][txn] Avoid creating multiple future and exception handlers [#15089](https://github.com/apache/pulsar/pull/15089)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 13]
```diff
@@ -0,0 +1,213 @@
...
+- [fix][broker] Fix potential NPE in Replicator [#15003](https://github.com/apache/pulsar/pull/15003)
+- [improve][broker] Add log when update namespace policies with error [#14850](https://github.com/apache/pulsar/pull/14850)
+- [improve][broker] Add publishRateLimitedTimes to topic metrics [#15739](https://github.com/apache/pulsar/pull/15739)
+- [improve][broker] Avoid using blocking calls for the async method ``checkTopicOwnership`` [#15023](https://github.com/apache/pulsar/pull/15023)
+- [improve][broker] Cancel offload tasks when managed ledger closed [#14545](https://github.com/apache/pulsar/pull/14545)
+- [improve][broker] Close connection if a ping or pong message cannot be sent [#15382](https://github.com/apache/pulsar/pull/15382)
+- [improve][broker] Configure DLog Bookie, Pulsar, and Admin clients via pass through config [#15818](https://github.com/apache/pulsar/pull/15818)
+- [improve][broker] Full-support ssl provider, ciphers and protocols for broker service and proxy service [#14569](https://github.com/apache/pulsar/pull/14569)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:81]:
```suggestion
- [improve][broker] Full-support SSL provider, ciphers, and protocols for broker service and proxy service [#14569](https://github.com/apache/pulsar/pull/14569)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 14]
```diff
@@ -0,0 +1,213 @@
...
+
+- [fix][auth] Add timeout of sync methods and  avoid call sync method for AuthoriationService [#15694](https://github.com/apache/pulsar/pull/15694)
+- [fix][auth] Fix debug log authenticate role error [#14784](https://github.com/apache/pulsar/pull/14784)
+- [fix][auth] Fix grant all permissions but can't list topic [#15501](https://github.com/apache/pulsar/pull/15501)
+- [fix][auth] Fix handling single role and non-jwt-token in MultiRolesTokenAuthorizationProvider [#14857](https://github.com/apache/pulsar/pull/14857)
+- [fix][auth] Generate correct well-known OpenID configuration URL [#15928](https://github.com/apache/pulsar/pull/15928)
+- [fix][auth] Role with namespace produce authz can also get topics [#15740](https://github.com/apache/pulsar/pull/15740)
+- [improve][auth] Add KeyStore support in WebSocket, Function Worker HTTPS Servers  [#15084](https://github.com/apache/pulsar/pull/15084)
+- [improve][auth] Allow to config web server's cipher and protocols [#13354](https://github.com/apache/pulsar/pull/13354)
+- [improve][auth] Full-support set ssl provider, ciphers and protocols [#13740](https://github.com/apache/pulsar/pull/13740)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:145]:
```suggestion
- [improve][auth] Full-support set SSL provider, ciphers, and protocols [#13740](https://github.com/apache/pulsar/pull/13740)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 15]
```diff
@@ -0,0 +1,213 @@
...
+- [fix][broker] Fix duplicated delayed messages when all consumers disconnect [#14740](https://github.com/apache/pulsar/pull/14740)
+- [fix][broker] Fix the pid occupied check when use pulsar-daemon start or stop process [#14701](https://github.com/apache/pulsar/pull/14701)
+- [fix][broker] Fix potential NPE in Replicator [#15003](https://github.com/apache/pulsar/pull/15003)
+- [improve][broker] Add log when update namespace policies with error [#14850](https://github.com/apache/pulsar/pull/14850)
+- [improve][broker] Add publishRateLimitedTimes to topic metrics [#15739](https://github.com/apache/pulsar/pull/15739)
+- [improve][broker] Avoid using blocking calls for the async method ``checkTopicOwnership`` [#15023](https://github.com/apache/pulsar/pull/15023)
+- [improve][broker] Cancel offload tasks when managed ledger closed [#14545](https://github.com/apache/pulsar/pull/14545)
+- [improve][broker] Close connection if a ping or pong message cannot be sent [#15382](https://github.com/apache/pulsar/pull/15382)
+- [improve][broker] Configure DLog Bookie, Pulsar, and Admin clients via pass through config [#15818](https://github.com/apache/pulsar/pull/15818)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:80]:
```suggestion
- [improve][broker] Configure DLog Bookie, Pulsar, and Admin clients via pass-through config [#15818](https://github.com/apache/pulsar/pull/15818)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 16]
```diff
@@ -0,0 +1,213 @@
...
+- [fix][broker] Fix MetadataStoreException$NotFoundException while doing topic lookup [#15633](https://github.com/apache/pulsar/pull/15633)
+- [fix][broker] Fix calculation in getNumberOfEntriesInStorage [#15627](https://github.com/apache/pulsar/pull/15627)
+- [fix][broker] Fix error log miss stack trace when create tenant fail [#14366](https://github.com/apache/pulsar/pull/14366)
+- [fix][broker] Fix resource group does not report usage [#15292](https://github.com/apache/pulsar/pull/15292)
+- [fix][broker] Fix the pid occupied check when use pulsar-daemon start or stop process [#15745](https://github.com/apache/pulsar/pull/15745)
+- [fix][broker] Fix duplicated delayed messages when all consumers disconnect [#14740](https://github.com/apache/pulsar/pull/14740)
+- [fix][broker] Fix the pid occupied check when use pulsar-daemon start or stop process [#14701](https://github.com/apache/pulsar/pull/14701)
+- [fix][broker] Fix potential NPE in Replicator [#15003](https://github.com/apache/pulsar/pull/15003)
+- [improve][broker] Add log when update namespace policies with error [#14850](https://github.com/apache/pulsar/pull/14850)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:75]:
```suggestion
- [improve][broker] Add log when updating namespace policies with error [#14850](https://github.com/apache/pulsar/pull/14850)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 17]
```diff
@@ -0,0 +1,213 @@
...
+- [fix][broker] Expose configurationMetadataStore and localMetadataStore [#15661](https://github.com/apache/pulsar/pull/15661)
+- [fix][broker] Filter the virtual NIC with relative path [#14829](https://github.com/apache/pulsar/pull/14829)
+- [fix][broker] Fix MetadataStoreException$NotFoundException while doing topic lookup [#15633](https://github.com/apache/pulsar/pull/15633)
+- [fix][broker] Fix calculation in getNumberOfEntriesInStorage [#15627](https://github.com/apache/pulsar/pull/15627)
+- [fix][broker] Fix error log miss stack trace when create tenant fail [#14366](https://github.com/apache/pulsar/pull/14366)
+- [fix][broker] Fix resource group does not report usage [#15292](https://github.com/apache/pulsar/pull/15292)
+- [fix][broker] Fix the pid occupied check when use pulsar-daemon start or stop process [#15745](https://github.com/apache/pulsar/pull/15745)
+- [fix][broker] Fix duplicated delayed messages when all consumers disconnect [#14740](https://github.com/apache/pulsar/pull/14740)
+- [fix][broker] Fix the pid occupied check when use pulsar-daemon start or stop process [#14701](https://github.com/apache/pulsar/pull/14701)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:73]:
```suggestion
- [fix][broker] Fix the pid occupied check when using pulsar-daemon start or stop process [#14701](https://github.com/apache/pulsar/pull/14701)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 18]
```diff
@@ -0,0 +1,213 @@
...
+- [fix][broker] Use dispatchRateLimiterLock to update dispatchRateLimiter [#15601](https://github.com/apache/pulsar/pull/15601)
+- [fix][broker] When skipping updating mark delete position, execute callback with executor to prevent deadlock [#15971](https://github.com/apache/pulsar/pull/15971)
+- [fix][broker] Expose configurationMetadataStore and localMetadataStore [#15661](https://github.com/apache/pulsar/pull/15661)
+- [fix][broker] Filter the virtual NIC with relative path [#14829](https://github.com/apache/pulsar/pull/14829)
+- [fix][broker] Fix MetadataStoreException$NotFoundException while doing topic lookup [#15633](https://github.com/apache/pulsar/pull/15633)
+- [fix][broker] Fix calculation in getNumberOfEntriesInStorage [#15627](https://github.com/apache/pulsar/pull/15627)
+- [fix][broker] Fix error log miss stack trace when create tenant fail [#14366](https://github.com/apache/pulsar/pull/14366)
+- [fix][broker] Fix resource group does not report usage [#15292](https://github.com/apache/pulsar/pull/15292)
+- [fix][broker] Fix the pid occupied check when use pulsar-daemon start or stop process [#15745](https://github.com/apache/pulsar/pull/15745)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:71]: Is this item duplicate with #14701 in Line73?

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 19]
```diff
@@ -0,0 +1,213 @@
...
+- [fix][broker] Fix race condition between timeout and completion  in `OpAddEntry` [#15233](https://github.com/apache/pulsar/pull/15233)
+- [fix][broker] Fix race condition in updating lastMarkDeleteEntry field [#15031](https://github.com/apache/pulsar/pull/15031)
+- [fix][broker] Fix rewind failed when ``redeliverUnacknowledgedMessages`` [#15046](https://github.com/apache/pulsar/pull/15046)
+- [fix][broker] Fix topic policy reader close bug [#14897](https://github.com/apache/pulsar/pull/14897)
+- [fix][broker] Fix typo in enum name and handle closing of the channel properly since writeAndFlush is asynchronous [#15384](https://github.com/apache/pulsar/pull/15384)
+- [fix][broker] Fix when nextValidLedger is null caused NPE [#13975](https://github.com/apache/pulsar/pull/13975)
+- [fix][broker] Fix wrong prompt exception when get non-persistent topic list without GET_BUDNLE permission [#14638](https://github.com/apache/pulsar/pull/14638)
+- [fix][broker] Fix wrong state for non-durable cursor [#14869](https://github.com/apache/pulsar/pull/14869)
+- [fix][broker] Fix wrong unit of NIC speed on linux [#15770](https://github.com/apache/pulsar/pull/15770)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:58]:
```suggestion
- [fix][broker] Fix wrong unit of NIC speed on Linux [#15770](https://github.com/apache/pulsar/pull/15770)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 20]
```diff
@@ -0,0 +1,213 @@
...
+- [fix][broker] Fix problem at RateLimiter#tryAcquire [#15306](https://github.com/apache/pulsar/pull/15306)
+- [fix][broker] Fix producerFuture not completed in ServerCnx#handleProducer [#14467](https://github.com/apache/pulsar/pull/14467)
+- [fix][broker] Fix race condition between timeout and completion  in `OpAddEntry` [#15233](https://github.com/apache/pulsar/pull/15233)
+- [fix][broker] Fix race condition in updating lastMarkDeleteEntry field [#15031](https://github.com/apache/pulsar/pull/15031)
+- [fix][broker] Fix rewind failed when ``redeliverUnacknowledgedMessages`` [#15046](https://github.com/apache/pulsar/pull/15046)
+- [fix][broker] Fix topic policy reader close bug [#14897](https://github.com/apache/pulsar/pull/14897)
+- [fix][broker] Fix typo in enum name and handle closing of the channel properly since writeAndFlush is asynchronous [#15384](https://github.com/apache/pulsar/pull/15384)
+- [fix][broker] Fix when nextValidLedger is null caused NPE [#13975](https://github.com/apache/pulsar/pull/13975)
+- [fix][broker] Fix wrong prompt exception when get non-persistent topic list without GET_BUDNLE permission [#14638](https://github.com/apache/pulsar/pull/14638)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:56]:
```suggestion
- [fix][broker] Fix wrong prompt exception when getting the non-persistent topic list without GET_BUDNLE permission [#14638](https://github.com/apache/pulsar/pull/14638)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md [hunk 21]
```diff
@@ -0,0 +1,213 @@
...
+- [fix][broker] Fix call sync method in onPoliciesUpdate method [#13885](https://github.com/apache/pulsar/pull/13885)
+- [fix][broker] Fix call sync method in onPoliciesUpdate method [#15227](https://github.com/apache/pulsar/pull/15227)
+- [fix][broker] Fix cannot delete namespace with system topic [#14730](https://github.com/apache/pulsar/pull/14730)
+- [fix][broker] Fix creating producer failure when set backlog quota [#15663](https://github.com/apache/pulsar/pull/15663)
+- [fix][broker] Fix creating system namespace topic failure [#14949](https://github.com/apache/pulsar/pull/14949)
+- [fix][broker] Fix deadlock in broker after race condition in topic creation failure [#15570](https://github.com/apache/pulsar/pull/15570)
+- [fix][broker] Fix getPendingAckInternalStats redirect issue [#14876](https://github.com/apache/pulsar/pull/14876)
+- [fix][broker] Fix inconsistent prompt message when schema version is empty using AVRO [#14626](https://github.com/apache/pulsar/pull/14626)
+- [fix][broker] Fix incorrect entryId in warn log when reading entry from tiered storage [#14685](https://github.com/apache/pulsar/pull/14685)
```
Comment (momo-jun) [inline site2/website-next/release-notes/versioned/pulsar-2.9.3.md:41]:
```suggestion
- [fix][broker] Fix incorrect entryId in warning logs when reading an entry from tiered storage [#14685](https://github.com/apache/pulsar/pull/14685)
```

File: site2/website-next/release-notes/versioned/pulsar-2.9.3.md (hunk 18 above)
Comment (mattisonchao) [reply site2/website-next/release-notes/versioned/pulsar-2.9.3.md:71]: Oh, thanks! I will fix it ASAP.

Comment (mattisonchao) [reply site2/website-next/release-notes/versioned/pulsar-2.9.3.md:71]: Fixed.

**Review by Anonymitaet — APPROVED** [review]

**Comment by mattisonchao** [comment]
@codelipenghui  PTAL, when you have time. Thanks!

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**